File: src/main/java/com/kesherManager/kesherManager/service/BoxService.java
package com.kesherManager.kesherManager.service;

import com.kesherManager.kesherManager.model.Box;
import com.kesherManager.kesherManager.model.Transport;

import java.util.List;
import java.util.Optional;

public interface BoxService {
    // Basic CRUD operations
    List<Box> getAllBoxes();

    Optional<Box> getBoxById(Long id);

    Box saveBox(Box box);

    void deleteBox(Long id);

    // Custom operations
    List<Box> getBoxesByStatus(Box.BoxStatus status);

    List<Box> searchBoxesByLocationName(String locationName);

    List<Box> searchBoxesByAddress(String address);

    // Update box status
    Box updateStatus(Long boxId, Box.BoxStatus newStatus);

    // Get transports related to a box
    List<Transport> getOutgoingTransports(Long boxId);

    List<Transport> getIncomingTransports(Long boxId);

    List<Box> searchBoxesByResponsiblePerson(String responsiblePerson);
    List<Box> searchBoxesByAssociationManager(String associationManager);
    List<Box> getBoxesByDonationGroup(String donationGroup);
    Box updateBoxDetails(Long boxId, Box boxDetails);

}

File: frontend/src/services/boxService.js
import api from './api';

const boxService = {
  // Get all boxes
  getAllBoxes: async () => {
    const response = await api.get('/api/boxes');
    return response.data;
  },

  // Get box by ID
  getBoxById: async (id) => {
    const response = await api.get(`/api/boxes/${id}`);
    return response.data;
  },

  // Create new box
  createBox: async (boxData) => {
    const response = await api.post('/api/boxes', boxData);
    return response.data;
  },

  // Update existing box
  updateBox: async (id, boxData) => {
    const response = await api.put(`/api/boxes/${id}`, boxData);
    return response.data;
  },

  // Delete box
  deleteBox: async (id) => {
    const response = await api.delete(`/api/boxes/${id}`);
    return response.data;
  },

  // Get boxes by status
  getBoxesByStatus: async (status) => {
    const response = await api.get(`/api/boxes/status/${status}`);
    return response.data;
  },

  // Get outgoing transports for a box
  getOutgoingTransports: async (boxId) => {
    const response = await api.get(`/api/boxes/${boxId}/outgoing-transports`);
    return response.data;
  },

  // Get incoming transports for a box
  getIncomingTransports: async (boxId) => {
    const response = await api.get(`/api/boxes/${boxId}/incoming-transports`);
    return response.data;
  }
};

export default boxService;
